src/Plugin/Action/AddOgMembershipRole.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Drupal\og\Plugin\Action;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Entity\OgRole;
+
+/**
+ * Adds a role to a user.
+ *
+ * @Action(
+ *   id = "og_membership_add_role_action",
+ *   label = @Translation("Add a role to the selected members"),
+ *   type = "og_membership"
+ * )
+ */
+class AddOgMembershipRole extends ChangeOgMembershipRoleBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute(OgMembership $member = NULL) {
+    if (!$member) {
+      return;
+    }
+    $rid = $this->configuration['rid'];
+    $roles = $member->getRoles();
```

Comment:
I think this can be simplified by changing from this line on to
`

    if (!in_array($rid, $member->getRolesIds())) {
      $member->addRole(OgRole::load($rid));
      $member->save();
    }
`
The og is already taking care of filtering out duplicated roles when adding one.